pkg/crypto/keys/publickey.go

```diff
@@ -34,6 +34,37 @@ func (keys PublicKeys) Less(i, j int) bool {
 	return keys[i].Y.Cmp(keys[j].Y) == -1
 }
 
+// DecodeBytes decodes a PublicKeys from the given slice of bytes.
+func (keys PublicKeys) DecodeBytes(data []byte) error {
+	b := io.NewBinReaderFromBuf(data)
+	b.ReadArray(keys)
+	return b.Err
+}
+
+// Contains checks whether passed param contained in PublicKeys.
+func (keys PublicKeys) Contains(pKey *PublicKey) bool {
+	for _, key := range keys {
+		if key == pKey {
+			return true
+		}
+	}
+	return false
+}
+
+// Unique returns set of public keys.
+func (keys PublicKeys) Unique() PublicKeys {
+	unique := make([]*PublicKey, 0, keys.Len())
+	pkMap := make(map[*PublicKey]bool)
+
+	for _, value := range keys {
+		if _, ok := pkMap[value]; !ok {
```

Comment:
We can just check the value of `pkMap[value]`

Reply:
even made better=)